source/class/qx/data/controller/CheckedList.js

```diff
@@ -0,0 +1,344 @@
+     */
+    __onTargetDetachResultsTag(evt) {
+      let { tagWidget, item } = evt.getData();
+      this.__detachTag(tagWidget, item);
+      item.setUserData(this.classname + ".tagWidget", null);
+    },
+    
+    /**
+     * Updates all tags in the target widget
+     */
+    __updateTags() {
+      let target = this.getTarget();
+      if (!target) {
+        return;
+      }
+      target.getChecked().forEach(item => {
+        let tagWidget = item.getUserData(this.classname + ".tagWidget");
+        this.__detachTag(tagWidget, item);
+        this.__attachTag(tagWidget, item);
+      });
+    },
+    
+    /**
+     * Attaches a single tag; used to bind to the tag so that live updates to the underlying model are reflected in tag names
+     *
+     * @param tagWidget {qx.ui.core.Widget} the widget which is the tag
+     * @param item {qx.ui.core.Widget} the list item that lists the model item that this tag is for
+     */
+    __attachTag(tagWidget, item) {
+      let itemModel = item.getModel();
+      let bindData = { };
+      if (this.getCheckedLabelPath()) {
+        bindData.checkedLabelId = itemModel.bind(this.getCheckedLabelPath(), tagWidget, "label", this.getCheckedLabelOptions());
+      }
+      if (this.getCheckedIconPath()) {
+        bindData.checkedIconId = itemModel.bind(this.getCheckedIconPath(), tagWidget, "label", this.getCheckedIconOptions());
+      }
+      itemModel.setUserData(this.classname + ".bindData", bindData);
+    },
+    
+    /**
+     * Detaches a single tag, inverse of `__attachTag`
+     *
+     * @param tagWidget {qx.ui.core.Widget} the widget which is the tag
+     * @param item {qx.ui.core.Widget} the list item that lists the model item that this tag is for
+     */
+    __detachTag(tagWidget, item) {
+      let itemModel = item.getModel();
+      let bindData = itemModel.getUserData(this.classname + ".bindData");
+      if (bindData) {
+        if (bindData.checkedLabelId) {
+          itemModel.removeListenerById(bindData.checkedLabelId);
+        }
+        if (bindData.checkedIconId) {
+          itemModel.removeListenerById(bindData.checkedIconId);
+        }
+        itemModel.setUserData(this.classname + ".bindData", null);
+      }
+    },
+      
+    /**
+     * Updates the checked widget items to match the array of checked model items
+     */
+    _updateChecked() {
+      let target = this.getTarget();
+      if (!target) {
+        return;
+      }
+      
+      if (this.__inUpdateChecked) {
+        return;
+      }
+      this.__inUpdateChecked = true;
+      try {
+        // Maps of the widget item, indexed by the hashcode of the model item      
+        let children = {};
+        let toUncheck = {};
+        
+        target.getChildren().forEach(item => {
+          let itemModel = item.getModel();
+          let hash = itemModel ? itemModel.toHashCode() : "null";
+          children[hash] = item;
```

Comment:
It appears possible for multiple items to hash to "null" thus overwriting items in the children map. Is this confirmed to be acceptable here?